lib/dbs/reports.js

```diff
@@ -0,0 +1,128 @@
+var createIndexer = require('level-simple-indexes')
+var levelPromise = require('level-promise')
+var EventEmitter = require('events')
+var assert = require('assert')
+var sublevel = require('subleveldown')
+var mtb36 = require('monotonic-timestamp-base36')
+var collect = require('stream-collector')
+var { promisifyModule } = require('../helpers')
+
+// exported api
+// =
+
+class ReportsDB extends EventEmitter {
+  constructor (cloud) {
+    super()
+    this.indexDB = sublevel(cloud.db, 'reports-index')
+
+    // create levels and indexer
+    this.reportsDB = sublevel(cloud.db, 'reports', { valueEncoding: 'json' })
+
+    this.indexer = createIndexer(this.indexDB, {
+      keyName: 'id',
+      properties: ['status', 'notes'],
+      map: (id, next) => {
+        this.getByID(id)
+          .catch(next)
+          .then(res => next(null, res))
+      }
+    })
+
+    // promisify
+    levelPromise.install(this.reportsDB)
+    levelPromise.install(this.indexDB)
+    promisifyModule(this.indexer, ['findOne', 'addIndexes', 'removeIndexes', 'updateIndexes'])
+  }
+
+  // basic ops
+  // =
+
+  async create (record) {
+    assert(record && typeof record === 'object')
+    assert(typeof record.archiveKey === 'string')
+
+    record = Object.assign({}, ReportsDB.defaults(), record)
+    record.id = `${record.archiveKey}:${mtb36()}`
+    record.createdAt = Date.now()
+
+    await this.put(record)
+    this.emit('create', record)
+    return record
+  }
+
+  async put (record) {
+    assert(typeof record.id === 'string')
+    record.updatedAt = Date.now()
+    await this.reportsDB.put(record.id, record)
```

Comment:
I should also point out that you don't have an `updateIndexes` call here, which you should if you're using indexes